docs/04-operating-scs/08-backups/user_data_backups.md

```diff
@@ -0,0 +1,303 @@
+   - `openstack volume create --snapshot $SNAPSHOT_NAME $TEMP_VOLUME_NAME`
+3. Wait until the volume creation is finished and the temporary volume reaches the `available` status.
+4. Create a backup image of the temporary volume:
+   - `openstack image create --volume $TEMP_VOLUME_NAME $IMAGE_NAME`
+5. Wait until the image creation finishes and the target image reaches the `active` status.
+6. Delete the temporary volume and snapshot:
+   - `openstack volume delete $TEMP_VOLUME_NAME`
+   - `openstack volume snapshot delete $SNAPSHOT_NAME`
+
+A full backup copy of the volume now resides in the Glance storage backend.
+
+If the necessity arises to store this backup outside of the IaaS infrastructure, the download procedure as described in [Image backup using download](#image-backup-using-download) may be used after the image creation.
+
+## Barbican secrets backup using download
+
+:::danger
+
+Secrets downloaded from Barbican will be in plaintext, which means that the secret is unprotected once received from the API.
+Before downloading secrets from Barbican make sure that a secure target environment is established for receiving and securely storing the secret's contents.
+
+:::
+
+Barbican secrets can be downloaded in plaintext using the corresponding API or client command:
+
+```bash
+openstack secret get --file $TARGET_FILE_PATH --payload_content_type "application/octet-stream" $SECRET_ID
+```
+
+### Retrieving encryption keys from Barbican
+
+In case of encrypted volumes (i.e. using a volume type with encryption), a corresponding encryption key is stored in Barbican.
+When an image is created from such a volume, the encryption key is duplicated in Barbican for the image.
+In order to backup those keys, the corresponding secret must first be identified.
+
+For volumes, this is possible starting with the Volume API microversion 3.64:
+
+```bash
+openstack volume show --os-volume-api-version 3.64 $VOLUME_NAME_OR_ID
+```
+
+The response will contain an `encryption_key_id` field with the ID of the Barbican secret.
+
+For images, the secret reference is stored in the `properties` field instead:
+
+```bash
+openstack image show -f value -c properties $IMAGE_NAME_OR_ID
+```
+
+In case of images created from encrypted volumes, the resulting output will have a nested `cinder_encryption_key_id` field that contains the ID of the Barbican secret.
+
+The resulting IDs can be used to retrieve the corresponding key using the [Barbican instructions](#barbican-secrets-backup-using-download) above.
+
+:::caution
+
+Note that the key retrieved from the secret is not immediately usable as LUKS passphrase to the image data of the volume.
+OpenStack does some processing to the key before it is passed to the LUKS encryption, which must be mimicked accordingly in order to unlock the encryption outside of OpenStack!
+
+See the [example procedure for converting the LUKS key](#luks-encryption-key-conversion-to-decrypt-volume-images) in the appendix section.
+
```

Comment:
Would it work, when you upload both: the image with similar metadata and the key into another OpenStack deployment?


I know that it most likely will fail with the image upload, but this is somehow interesting.

Reply:
Very intriguing point. I haven't thought about that yet. I could see this working via `openstack secret store ...` + `openstack image create --property cinder_encryption_key_id=$SECRET_ID ...`.

I will test that.

On that note... should the guide also cover backup restore procedures aside from manually cracking open the LUKS dump locally as currently illustrated in the appendix? I guess it would be a good idea now that you bring it up.

Reply:
Well I see this use case:
1. Download image from one deployment (maybe encrypted)
1.1. Download associated key
2. Upload Image to another deployment either as backup or already ready to use
2.1. Upload associated key

as more common than storing a downloaded image privately.

Describing a tested and safe way for that would be also good I think.

Reply:
I have now included restore instructions addressing all of this.